### PR TITLE
focuswriter: 1.7.6 -> 1.8.5

### PR DESCRIPTION
**Pull request by kashw2**
## Description of changes

change log: https://github.com/gottcode/focuswriter/releases/tag/v1.8.5

diff: https://github.com/gottcode/focuswriter/compare/v1.7.6...v1.8.5

- added myself as a maintainer of focuswriter
- changed `all-packages.nix` call to focuswriter to use `qt6Packages.callPackage` over `libsForQt5.callPackage` as focuswriter now uses qt6 predominantly. 
- swapped out `fetchurl` to `fetchFromGitHub` as the GitHub acts as a sort of mirror and has interop with `nix-update` for ease of updates in the future
- wrapped application

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by kashw2** [comment]
Result of `nixpkgs-review pr 253111` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>focuswriter</li>
  </ul>
</details>

**Review by NickCao — CHANGES_REQUESTED** [review]
Hash is not updated, also please split the maintainer addition into its own commit.

**Comment by kashw2** [comment]
Hi @NickCao. Thank you for catching this.

I've actually been given a good opportunity to learn more about qt6 and packaging qt apps for Nix due to this which I appreciate. 

I've added a few more commits. I've updated the description of the Pull Request to document the changes and their rationales.

Thanks.

**Comment by kashw2** [comment]
Result of `nixpkgs-review pr 253111` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>focuswriter</li>
  </ul>
</details>

File: pkgs/applications/editors/focuswriter/default.nix [hunk 1]
```diff
@@ -1,24 +1,37 @@
...
+    rev = "v${version}";
+    hash = "sha256-6wvTlC/NCCcN2jpwqtoOsCln3ViY/vj7NpMsbYHBGiI=";
   };
 
-  nativeBuildInputs = [ pkg-config qmake qttools ];
-  buildInputs = [ hunspell qtbase qtmultimedia ];
+  nativeBuildInputs = [ pkg-config cmake qttools ];
+  buildInputs = [ hunspell qtbase qtmultimedia qt5compat ];
+
+  dontWrapQtApps = true;
```
Comment (NickCao) [inline pkgs/applications/editors/focuswriter/default.nix:27]: Why aren't we wrapping it?

File: pkgs/applications/editors/focuswriter/default.nix [hunk 2]
```diff
@@ -1,24 +1,37 @@
...
+    owner = "gottcode";
+    repo = "focuswriter";
+    rev = "v${version}";
+    hash = "sha256-6wvTlC/NCCcN2jpwqtoOsCln3ViY/vj7NpMsbYHBGiI=";
   };
 
-  nativeBuildInputs = [ pkg-config qmake qttools ];
-  buildInputs = [ hunspell qtbase qtmultimedia ];
+  nativeBuildInputs = [ pkg-config cmake qttools ];
+  buildInputs = [ hunspell qtbase qtmultimedia qt5compat ];
```
Comment (NickCao) [inline pkgs/applications/editors/focuswriter/default.nix:25]: Also please add `qtwayland` for, wayland support.

File: pkgs/applications/editors/focuswriter/default.nix (hunk 1 above)
Comment (kashw2) [reply pkgs/applications/editors/focuswriter/default.nix]: I'd had issues early on when attempting to wrap it however it looks like those are gone now.

fixed in https://github.com/NixOS/nixpkgs/pull/253111/commits/1f95a84bdaff12b56bcb653c9251dba6206c8f70

File: pkgs/applications/editors/focuswriter/default.nix (hunk 2 above)
Comment (kashw2) [reply pkgs/applications/editors/focuswriter/default.nix]: added qtwayland to `buildInputs` in https://github.com/NixOS/nixpkgs/pull/253111/commits/1f95a84bdaff12b56bcb653c9251dba6206c8f70

**Review by NickCao — APPROVED** [review]
diff lgtm

**Comment by NickCao** [comment]
Tested building, running and typing a few words.

**Pull request merged**